libbeat/docs/shared-autodiscover.asciidoc

```diff
@@ -114,6 +114,57 @@ contain variables from the autodiscover event. They can be accessed under data n
 
 include::../../{beatname_lc}/docs/autodiscover-kubernetes-config.asciidoc[]
 
+[float]
+===== Jolokia (experimental)
+
+The Jolokia autodiscover provider uses Jolokia Discovery to find agents running
+in your host or your network.
+
+Jolokia Discovery mechanism is supported by any Jolokia agent since version
+1.2.0, it is enabled by default when Jolokia is included in the application as
+a JVM agent, but disabled in other cases as the OSGI or WAR (Java EE) agents.
+In any case, this feature is controlled with two properties:
+
+  * `discoveryEnabled`, to enable the feature
+  * `discoveryAgentUrl`, if set, this is the URL announced by the agent when
+    being discovered, setting this parameter implicitely enables the feature
+
+There are multiple ways of setting these properties, and they can vary from
+application to application, please refer to the documentation of your
+application to find the more suitable way to set them in your case.
+ 
+Jolokia Discovery is based on UDP multicast requests. Agents join the multicast
+group 239.192.48.84, port 24884, and discovery is done by sending queries to
+this group. You have to take into account that UDP traffic between Metricbeat
+and the Jolokia agents has to be allowed. Also notice that this multicast
+address is in the 239.0.0.0/8 range, that is reserved for private use within an
+organization, so it can only be used in private networks.
+
+These are the available fields on every event:
+
+  * jolokia.agent.id
```

Comment:
Are these fields also added to each event indirectly created through auto discovery?

Reply:
Yes, they are added, I think that by the Autodiscover framework.

Reply:
Everything under `meta` here is added: https://github.com/elastic/beats/pull/7141/files#diff-9132fdc5edac3dbc7a2ef1540ad3e471R68